ulid/__init__.py

```diff
@@ -22,13 +22,17 @@
 __version__ = version("python-ulid")
 
 
-class validate_type:  # noqa: N801
-    def __init__(self, *types: Any) -> None:
+T = TypeVar("T", bound=type)
+ReturnType = TypeVar("ReturnType")
+
+
+class validate_type(Generic[T]):  # noqa: N801
```

Comment:
Isn't this generic over `T` **and** `ReturnType`?

Reply:
I may be wrong, but don't think so.  If we made it `validate_type(Generic[T, ReturnType])`, I think we would need to take in both the `T` and `ReturnType` in the `__init__`, which I don't think we want or need to do, e.g.:
```
class validate_type(Generic[T]):
    def __init__(self, return_type: ReturnType, *types: T):
        ...
```

The `ReturnType` isn't being pinned when the decorator is run, it is simply indicating the wrapper will return the same type as the decorated method.